src/Framework/AssemblyLoadBuildEventArgs.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#nullable disable
+
+using System;
+using System.IO;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Framework
+{
+    public class AssemblyLoadBuildEventArgs : BuildMessageEventArgs
+    {
+        public AssemblyLoadBuildEventArgs()
+        { }
+
+        public AssemblyLoadBuildEventArgs(
+            AssemblyLoadingContext loadingContext,
+            string assemblyName,
+            string assemblyPath,
+            Guid mvid,
+            int appDomainId,
+            string appDomainFriendlyName,
+            MessageImportance importance = MessageImportance.Low)
+            : base(null, null, null, importance, DateTime.UtcNow, assemblyName, assemblyPath, mvid)
+        {
+            LoadingContext = loadingContext;
+            AssemblyName = assemblyName;
+            AssemblyPath = assemblyPath;
+            MVID = mvid;
+            AppDomainId = appDomainId;
+            AppDomainFriendlyName = appDomainFriendlyName;
+        }
+
+        public AssemblyLoadingContext LoadingContext { get; private set; }
+        public string AssemblyName { get; private set; }
+        public string AssemblyPath { get; private set; }
+        public Guid MVID { get; private set; }
+        public int AppDomainId { get; private set; }
+        public string AppDomainFriendlyName { get; private set; }
```

Comment:
Think it's worth anything to avoid storing and reporting these for the default app domain and/or on core where there are no app domains?

Would indicating `AssemblyLoadContext` on core be helpful at all (I don't think so at the moment)?

Reply:
Fair enough - updating to store/transfer null (single bit) for cases where `IsDefaultAppDomain()` returns `true`.
This handles core and framework and is prepared for cases like running task on core, but reporting the result to framework scheduler (not currently possible - but might be in the future) or vice versa; while using single bit of wire format for the most common cases.